mesonbuild/compilers/mixins/clike.py

```diff
@@ -764,7 +869,7 @@ def has_function(self, funcname, prefix, env, *, extra_args=None, dependencies=N
             extra_args = []
 
         # Short-circuit if the check is already provided by the cross-info file
-        varname = 'has function ' + funcname
+        varname = 'has function ' + typename
```

Comment:
Hmm, why are we renaming from funcname to typename in this function?

Reply:
Because typename is what the base compiler calls it and I just copied the signature from there and touched up the function body. I can change then to funcname if you prefer

Reply:
I think it should be `funcname` because that's actually what it is, the base compiler prototype should be changed here :)